From 81d1bc7f01729ca28419213f2782860b38758d29 Mon Sep 17 00:00:00 2001
From: Jonathon Hall <jhall@londontrustmedia.com>
Date: Fri, 8 Mar 2019 03:03:42 +0900
Subject: [PATCH 09/10] Accept other TAP adapter IDs as well

---
 src/openvpn/tun.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/openvpn/tun.c b/src/openvpn/tun.c
index 63f9d1bf..065b4715 100644
--- a/src/openvpn/tun.c
+++ b/src/openvpn/tun.c
@@ -3701,6 +3701,8 @@ get_tap_reg(struct gc_arena *gc)
                 if (status == ERROR_SUCCESS && data_type == REG_SZ)
                 {
                     if (!strcmp(component_id, TAP_WIN_COMPONENT_ID) ||
+                        !strcmp(component_id, "tap-pia-0901") ||
+                        !strcmp(component_id, "root\\tap-pia-0901") ||
                         !strcmp(component_id, "root\\" TAP_WIN_COMPONENT_ID))
                     {
                         struct tap_reg *reg;
-- 
2.22.0

