From ff1fbfd10754c29e27de4c90afb0a286a7f947af Mon Sep 17 00:00:00 2001
From: Jonathon Hall <jhall@londontrustmedia.com>
Date: Thu, 21 Mar 2019 09:59:56 -0400
Subject: [PATCH 2/5] Support 32-bit builds in build.bat

---
 msvc/build.bat | 29 ++++++++++++++++++++---------
 1 file changed, 20 insertions(+), 9 deletions(-)

diff --git a/msvc/build.bat b/msvc/build.bat
index 153cfd6..59441be 100644
--- a/msvc/build.bat
+++ b/msvc/build.bat
@@ -35,12 +35,23 @@ SET OPENVPN_BUILD_OPENVPN=openvpn-build-openvpn
 call build-env.bat
 if exist build-env-local.bat call build-env-local.bat
 
+if "%BITS%" == "32" (
+    set VCVARS_ARCH=x86
+    set OPENSSL_ARCH=VC-WIN32
+    set OPENVPN_ARCH=Win32
+) else (
+    set BITS=64
+    set VCVARS_ARCH=x64
+    set OPENSSL_ARCH=VC-WIN64A
+    set OPENVPN_ARCH=x64
+)
+
 if exist "%VCHOME%\vcvarsall.bat" (
-	call "%VCHOME%\vcvarsall.bat"
-) else if exist "%VCHOME%\bin\vcvars64.bat" (
-	call "%VCHOME%\bin\vcvars64.bat"
-) else if exist "%VCHOME%\Auxiliary\Build\vcvars64.bat" (
-	call "%VCHOME%\Auxiliary\Build\vcvars64.bat"
+	call "%VCHOME%\vcvarsall.bat" %VCVARS_ARCH%
+) else if exist "%VCHOME%\bin\vcvars%BITS%.bat" (
+	call "%VCHOME%\bin\vcvars%BITS%.bat"
+) else if exist "%VCHOME%\Auxiliary\Build\vcvars%BITS%.bat" (
+	call "%VCHOME%\Auxiliary\Build\vcvars%BITS%.bat"
 ) else (
 	echo Cannot detect visual studio environment
 	goto error
@@ -113,7 +124,7 @@ if errorlevel 1 goto error
 echo Build OpenSSL
 
 cd build.tmp\openssl*
-perl Configure VC-WIN64A --prefix="%TARGET%" --openssldir="%TARGET%"\ssl
+perl Configure %OPENSSL_ARCH% --prefix="%TARGET%" --openssldir="%TARGET%"\ssl
 if errorlevel 1 goto error
 nmake install
 if errorlevel 1 goto error
@@ -129,7 +140,7 @@ cd %ROOT%
 echo Build LZO
 
 cd build.tmp\lzo*
-call B\win64\vc_dll.bat
+call B\win%BITS%\vc_dll.bat
 rem if errorlevel 1 goto error - returns 1!!
 xcopy include\lzo "%TARGET%\include\lzo" /e /i /y
 if errorlevel 1 goto error
@@ -173,9 +184,9 @@ mkdir ..\..\%OPENVPN_BUILD_OPENVPN% > nul 2>&1
 cd build.tmp\openvpn*
 xcopy * ..\..\..\..\%OPENVPN_BUILD_OPENVPN% /E
 cd ..\..\..\..\%OPENVPN_BUILD_OPENVPN%
-msbuild openvpn.sln /p:Platform=x64 /p:Configuration=%RELEASE%
+msbuild openvpn.sln /p:Platform=%OPENVPN_ARCH% /p:Configuration=%RELEASE%
 if errorlevel 1 goto error
-copy x64-Output\%RELEASE%\*.exe "%TARGET%\bin"
+copy %OPENVPN_ARCH%-Output\%RELEASE%\*.exe "%TARGET%\bin"
 if errorlevel 1 goto error
 copy include\openvpn-*.h "%TARGET%\include"
 if errorlevel 1 goto error
-- 
2.23.0.windows.1

